src/relay/pass/device_annotation.cc

```diff
@@ -176,7 +176,11 @@ class RewriteAnnotation : public ExprMutator {
   }
 
   Expr VisitExpr_(const CallNode* call_node) final {
-    if (IsOnDeviceNode(call_node) || IsDeviceCopyNode(call_node)) {
+    if (IsOnDeviceNode(call_node)) {
```

Comment:
This is the only meaningful change right? this will just search deeper in the tree to rewrite until we hit another device annotation? just want to make sure 100% understand this time.

Reply:
yes, for example, `on_device` connects `a` and `b` expression, `a -> on_device -> b`, we have inserted `device_copy` when we visit `b`. Anytime when we visit `on_device`, if means we either have inserted `device_copy` op, or it is not necessary, therefore, we can safely delete `on_device` by returning '`a`.

Reply:
Okay cool, LGTM